tests/e2e/010_interfaces_test.go

```diff
@@ -98,13 +98,13 @@ func TestInterfaceConnTap(t *testing.T) {
 
 	// restart microservice twice
 	for i := 0; i < 2; i++ {
-		ctx.StopMicroservice(msName)
+		ctx.Expect(ms.Stop()).To(Succeed())
```

Comment:
Could this check happen inside the `Stop()` function. This call is operating on the test setup and is not really part of this test case - just part of the test framework needed to run the test so it makes more sense to make the error checking in the background.

Reply:
Agreed, checking the error inside the test itself isn't very elegant. The problem is the same function is used inside the test setup/harness itself, where returning error and then checking it is potentially useful (agent.Stop() works the same and can return different errors). After test cases and test harness are separated into different packages, the solution could be to have unexported  stop() function that returns an error (for use inside the test harness package) and then one exported function that doesn't return an error (for use in test cases). 
For now in the test cases we could: 

- revert to using ctx.StopMicroservice 
- not check the error at all 
- leave it as is

The first solution seems the best to me until the split happens in separate PR.